tests/unit_tests/test_config.py

```diff
@@ -146,4 +150,91 @@ def test_auth_from_env_throws_when_not_available():
+    """
+    for key, value in subset.items():
+        if key not in superset:
+            return False
+        superset_value = superset[key]
+
+        # If both values are dictionaries, recurse
+        if isinstance(value, dict) and isinstance(superset_value, dict):
+            if not is_subset(value, superset_value):
+                return False
+        # Check equality for non-dict values, ignoring None in superset
+        elif superset_value is not None and value != superset_value:
+            return False
+    return True
+
+
+# Parameterize the fixture to accept different config examples
+@pytest.fixture
+def temp_yaml_config_file(request):
+    # Use the provided config data from test parameters
+    config_data = request.param
+
+    # Create a temporary YAML file with the configuration
+    with tempfile.NamedTemporaryFile(
+        suffix=".yaml", mode="w", delete=False
+    ) as temp_yaml_file:
+        yaml.dump(config_data, temp_yaml_file)
+        temp_yaml_file_path = temp_yaml_file.name
+
+    # Provide the path and the config data
+    yield Path(temp_yaml_file_path), config_data
+
+    # Cleanup after test execution
+    os.remove(temp_yaml_file_path)
+
+
+# Parameterized test to run with different configurations
+@pytest.mark.parametrize(
+    "temp_yaml_config_file",
+    [
+        # Different configuration examples passed to the fixture
+        {
+            "env": {
+                "sources": [
+                    {"kind": "dodal", "module": "dodal.adsim"},
+                    {"kind": "planFunctions", "module": "dls_bluesky_core.plans"},
+                    {"kind": "planFunctions", "module": "dls_bluesky_core.stubs"},
+                ],
+            },
+            "api": {"host": "0.0.0.0", "port": 8000},
+        },
+        {
+            "stomp": None,
+            "env": {
+                "sources": [
+                    {"kind": "dodal", "module": "dodal.adsim"},
+                    {"kind": "planFunctions", "module": "dls_bluesky_core.plans"},
+                    {"kind": "planFunctions", "module": "dls_bluesky_core.stubs"},
+                ],
+                "events": {"broadcast_status_events": True},
+            },
+            "logging": {"level": "INFO"},
+            "api": {"host": "0.0.0.0", "port": 8000, "protocol": "http"},
+            "scratch": None,
+        },
+    ],
+    indirect=True,
+)
+def test_config_yaml_parsed(temp_yaml_config_file):
+    temp_yaml_file_path, config_data = temp_yaml_config_file
+
+    # Initialize loader and load config from the YAML file
+    loader = ConfigLoader(ApplicationConfig)
+    loader.use_values_from_yaml(temp_yaml_file_path)
+    loaded_config = loader.load()
+
+    # Parse the loaded config JSON into a dictionary
+    target_dict_json = json.loads(loaded_config.model_dump_json())
+
+    # Assert that config_data is a subset of target_dict_json
+    assert is_subset(config_data, target_dict_json), ""
```

Comment:
Could: Asserting for a subset is of limited value, it won't detect some old parameter whose name has changed still being in the config, for example. You could assume the test config is exhaustive and assert equals.

Reply:
> , it won't detect some old parameter whose name has changed still being in the config, for example


not sure when would that happen.


the yaml comfig may well be underspecified compared to the whole config space, such as leaving out `api` empty, or `stomp` empty. that is why the subset is used

Reply:
Hmm, yes I see your point, there are actually two possible tests here:
1. Take an exhaustive config and check that it is valid and that it really is exhaustive
2. Take a non-exhaustive config and check that it is valid
This is 2.

Do we think it's worth writing both? I think 1. would be useful

Reply:
ok will write the test for

Reply:
got a red test, now making it green